pkg/velero/storagelocation/paths/paths.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package paths  helps NAC controllers with getting relevant object store paths of interests
+// generally as defined in https://github.com/vmware-tanzu/velero/blob/0a280e57863c70495a2dfb65787615a68a0e7b03/pkg/persistence/object_store_layout.go
+package paths
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// BasePathForLocation returns base path for a given storage location
+// <protocol>://<bucket>/<prefix>
+func BasePathForLocation(storageLocation *velerov1.BackupStorageLocation) (basepath string, err error) {
+	var protocol string
+	switch storageLocation.Spec.Provider {
+	case "aws":
+		protocol = "s3://"
+	// TODO: implement azure
```

Comment:
the plan is to implement this prior to first release?

Reply:
Yeah we need to support all the default object storage providers